### PR TITLE
feat: simple Product model

### PR DESCRIPTION
**Pull request by raphodn**
### What

- new `Product` model & schema
- it's goal is to store metadata fetched from OpenFoodFacts, and link it to the Prices (next PRs)

### CODE REVIEW

File: app/enums.py [hunk 1]
```diff
@@ -1,6 +1,13 @@
 from enum import Enum
 
 
+class ProductOFFSource(Enum):
```
Comment (raphael0202) [inline app/enums.py:4]: There is the `Flavor` enum in the Python SDK just for that

File: app/schemas.py [hunk 2]
```diff
@@ -16,6 +23,22 @@ class UserBase(BaseModel):
...
 
+class ProductCreate(BaseModel):
+    model_config = ConfigDict(from_attributes=True, arbitrary_types_allowed=True)
+
+    code: str
+
+
+class ProductBase(BaseModel):
+    id: int
+    off_source: ProductOFFSource | None
```
Comment (raphael0202) [inline app/schemas.py:34]: why do you prefix with off? I don't think that the fact that off is the source is relevant, we can probably skip it :)

File: app/schemas.py [hunk 3]
```diff
@@ -16,6 +23,22 @@ class UserBase(BaseModel):
...
+    off_name: str | None
```
Comment (raphael0202) [inline app/schemas.py:35]: `product_name` maybe? To follow off field names

**Review by raphael0202 — APPROVED** [review]

File: app/enums.py (hunk 1 above)
Comment (raphodn) [reply app/enums.py:4]: ok awesome !

File: app/schemas.py (hunk 2 above)
Comment (raphodn) [reply app/schemas.py:34]: if we were to have other sources to get product info ?

File: app/schemas.py (hunk 3 above)
Comment (raphodn) [reply app/schemas.py:35]: yeah you're right, i'll do the same for `off_product_quantity`

File: app/schemas.py (hunk 2 above)
Comment (raphodn) [reply app/schemas.py:34]: Ok I'll remove the `off_` prefix :ok_hand:

**Pull request merged**